segmented_sieve_of_eratosthenes/segmentedSieveOfEratosthenes.cpp

```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+#include <cmath>
+#include <cstring>
+#include <vector>
+using namespace std;
+
+vector<int> Prime; //contains prime numbers uptop segMax
+void segmentedSieve(){
+	#define segMax 1000000 //till point you want to find Primes. Preferebly pow(10,x)
+	#define rootSegMax 1000 //root of segMax
+	bool prime[rootSegMax];
+	memset(prime, true, sizeof(prime));
+	for(int i = 2; i*i <= rootSegMax; i++){
+		if(prime[i]) {
+			for(int j = i*i; j < rootSegMax; j += i){
+				prime[j] = false;
+			}
+		}
+	}
+	for(int i = 2; i < rootSegMax; i++){
+		if(prime[i]) {
+			Prime.push_back(i);
+		}
+	}
+	int low = rootSegMax;
+	// ^^ lower end of current block we are finding primes for
+	int high = 2*rootSegMax;
+	// ^^ higher end of current block we are finding primes for
+	int tempMax = Prime.size();
+	// ^^All non prime numbers will be multiples of these numbers only
+	while(low < segMax){
```

Comment:
Keep a space between `(` and reserved keywords that use them. 
`while(` -> `while (`.

Reply:
I don't think it should be necessary. :)